src/R/Components/Impl/Repl/Commands/SourceRScriptCommand.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Common.Core.Shell;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.R.Components.Controller;
+using Microsoft.R.Components.Extensions;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.R.Package.Repl.Commands {
+    public sealed class SourceRScriptCommand : IMenuCommand {
+        private readonly IRInteractiveWorkflow _interactiveWorkflow;
+        private readonly IActiveWpfTextViewTracker _activeTextViewTracker;
+        private readonly bool _echo;
+
+        public SourceRScriptCommand(
+            IRInteractiveWorkflow interactiveWorkflow,
+            IActiveWpfTextViewTracker activeTextViewTracker,
+            bool echo
+        ) {
+            _interactiveWorkflow = interactiveWorkflow;
+            _activeTextViewTracker = activeTextViewTracker;
+            _echo = echo;
+        }
+
+        public CommandStatus Status {
+            get {
+                var status = CommandStatus.Supported;
+                if (_interactiveWorkflow.ActiveWindow == null) {
+                    status |= CommandStatus.Invisible;
+                } else if (!string.IsNullOrEmpty(GetFilePath())) {
+                    status |= CommandStatus.Enabled;
+                }
+                return status;
+            }
+        }
+
+        public CommandResult Invoke(object inputArg, ref object outputArg) {
```

Comment:
Renaming is suggested because the existing interface occupies the "ICommand" name that would really be more appropriate for the new thing (and even though they're in different namespaces, a lot of code that uses the interface needs both of them, causing conflicts). The proposed approach is a simple mechanical renaming of existing ICommand to IEditorCommand (or something along these lines), so it would be a simple Refactor -> Rename, not requiring any particular effort.

---
In reply to: [63592100](https://github.com/Microsoft/RTVS/pull/1785#discussion_r63592100) [](ancestors = 63592100)